### ISSUE TITLE
Regarding variable weights and normalisation statistics for fine-tuning

### ISSUE THREAD

**Issue by hwhitbymet** [microsoft/aurora#156]
I'm looking to fine-tune Aurora with some new variables (and extending existing variables with previously out-of-sample data). I understand that, to do this, I'll need to determine a weighting for each variable (used in the loss function) as well as normalisation statistics for each. I have a few questions regarding these - I've browsed the issues page here on GitHub and do not believe these have already been answered, but apologies if I missed something. My primary sources are the fine-tuning page on the docs (https://microsoft.github.io/aurora/finetuning.html), the latest version of the Aurora paper, and this GitHub repo.

Regarding per-variable weights in the loss function:  
Per the Aurora paper, the weight values used during pre-training are inspired by those used in Bi et al., 2023. In this paper, the authors performed an early training run "designed to facilitate equivalence of the contributions [to the loss value]" for each variable, the final weight value being inversely proportionate to that variable's average contribution to the loss value during that run. 

The weight values shared in the Aurora paper are slightly different to those from Bi et al., 2023, and I understand the Aurora authors have also separately extended the model with air pollutants and wave-ocean variables. Given this, I assume the Aurora authors performed their own initial training run(s) similar to Bi. et al, 2023, to determine the values used for each variable. My questions are as follows:

1. Can you share any insight regarding this initial training run? Details such as the number of epochs used, any specific pre-processing / preparation, etc. would be appreciated.

2. I understand that the final weight values are inversely proportionate to the loss generated by each variable in the early training run. However, the methodology used to determine the exact values is unclear - can you share any further insight on this? For example, "z" (geopotential/orography) has a weight value of 3.0 - exactly how did the Aurora authors arrive at this value?

3. All of the weights for the 'base' Aurora model fall within a range of 0 - 3 (or more precisely, 0.6 - 3). Were the final values normalised along this scale, or post-processed in some other way? Or is this a coincidence? If the values were normalised to fit a scale, are they placed linearly along this scale, or are they weighted somehow?

Regarding normalisation statistics:
I want to ensure that my methodology for calculating the normalisation statistics is equivalent to that used for Aurora's pre-training. While the values for these statistics used while training Aurora are not shared in the Aurora paper, I am aware there is a file in this GitHub repo `aurora/normalisation.py` containing what appear to be the final location (mean) / scale (standard deviation) statistics for each variable. I've been comparing my own mean / standard deviation values for all ERA5 variables used for Aurora's pre-training to the values from this file. My questions are as follows:

1. Table C.3 in the Aurora paper summarises the exact timeframes used for each training dataset, with ERA5 given the timeframe 1979-2020. I understand, however, that 2020 IFS-HRES data was used for validating the model during pre-training. Typically, I expect validation years not to overlap with those used in training - i.e., I would assume that the 1979-2020 timeframe given for ERA5 is exclusive of 2020. However, the use of a different dataset for validation gives me pause, as - even if you did include 2020 ERA5 data during training - this would still be unseen data and thus suitable to use as a validation split. Can you clarify this ambiguity, confirming whether the 1979-2020 time range used for the training dataset is inclusive or exclusive of 2020?

2. I have found that the location value in `normalisation.py` for the variable "z" (geopotential/orography) more closely correlates with the minimum value I have observed in the ERA5 data, as opposed to the standard deviation as would be expected. I can make a guess as to why this is, but I would like to be sure - is it the case that the Aurora authors used the minimum value rather than the standard deviation as the location statistic for this particular variable, and if so, why?

Thank you for your time.

References:
Bi, K., Xie, L., Zhang, H., Chen, X., Gu, X., Tian, Q.: Accurate medium-range global weather forecasting with 3d neural networks. Nature 619(7970), 533–538 (2023) https://doi.org/10.1038/s41586-023-06185-3

**Comment by wesselb** [comment]
Thanks for opening an issue, @hwhitbymet! :) I'll get back to you shortly.

**Comment by wesselb** [comment]
> Given this, I assume the Aurora authors performed their own initial training run(s) similar to Bi. et al, 2023, to determine the values used for each variable. 

We actually did not do an initial training run to determine the weights. For the base model, the weights were taken from Bi. et al (2023) and then slightly adjusted in an ad-hoc manner according to what the model struggled with most. For the air pollution and ocean wave experiments, the weights were also determined in an ad-hoc way according to which variables we expected to be the most difficult.

> All of the weights for the 'base' Aurora model fall within a range of 0 - 3 (or more precisely, 0.6 - 3). Were the final values normalised along this scale, or post-processed in some other way? 

Note that the loss is computed over the _normalised variables_. Hence, for a model that outputs zero, the squared error loss is should be one on average for all variables, meaning that all losses are perfectly balanced and no weighting would be necessary, which is equivalent to weights equal to 1. In practice, we used an MAE loss and the model of course doesn't output zero, so some adjustment of the weights away from 1 is necessary. This is the reason that most weights fall into the range [0, 3], which is centered around 1.

>  Typically, I expect validation years not to overlap with those used in training. (...) Can you clarify this ambiguity, confirming whether the 1979-2020 time range used for the training dataset is inclusive or exclusive of 2020?

The range is inclusive of 2020, meaning that the validation year indeed overlaps with the last pretraining year. I should double check this, but I believe that's the case. The validation data source (HRES T0), however, is not included in the pretraining mix, which should prevent overfitting. Note  that all numbers in the paper are computed over the test years, which are completely disjoint from the training years. Validation was only used to guide model development.

> I can make a guess as to why this is, but I would like to be sure - is it the case that the Aurora authors used the minimum value rather than the standard deviation as the location statistic for this particular variable, and if so, why?

The normalisation is of the following form: `(x - location) / scale`. By setting `location = minimum` and `scale = maximum - minimum`, this becomes `(x - minimum) / (maximum - minimum)`, i.e. normalisation to the interval [0, 1]. :)
